plugins/single_plugins/scale.cpp

```diff
@@ -263,7 +264,8 @@ class wayfire_scale : public wf::plugin_interface_t
             input_release_impending = false;
         }
 
-        if (button != BTN_LEFT || state != WLR_BUTTON_PRESSED)
+        if ( (button != BTN_LEFT && (button != BTN_MIDDLE ||
```

Comment:
Since there is more than one possible case now, it should be cased out like in `process_key()`, instead of a complex conditional.